### PR TITLE
VerticalStore foundations

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR introduces `VerticalStore`, `VerticalAction`, `VerticalSegmentModel`, `VerticalRestClient` and `VerticalSqlUtils`. Since we are still ironing out the details of this project, most of this is subject to change and this code is only being added as a foundation. Although there shouldn't be any conflicts, since this code will be changing frequently and will include a lot of todos, we'll use a master branch to merge into. This way we can ensure we address every `TODO` before we merge it into `develop`.

Speaking of `TODO`s, I've added a lot of them for things that I didn't want to forget about. Some of them, I do actually have the answers for, but I want to keep them for bookkeeping until we know more about the project.

Although I like to add tests as soon as possible, I can't justify the time I'll need to spend adding and updating them as we figure more about the project. However, as soon as we have most of it figured out, I'll be open a PR that adds tests all the necessary tests.

@malinajirka Let me know if you have any questions/concerns about the direction.

### CODE REVIEW

**Comment by malinajirka** [comment]
Thanks @oguzkocer. I have just scanned the code as it's subject to change. 

- My only concern is whether storing the data into a database isn't an overkill. As I believe in most cases the user creates just one site and storing the data increases the complexity of the code. Can't we get away with just an OkHttp cache?  -> it would also significantly reduced complexity of pagination. Wdyt?

**Comment by oguzkocer** [comment]
@malinajirka I've removed the DB layer in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/963/commits/8df94afedd2d293299dac1e3caacf09503dcbb72 and added the fetched segments to the `OnSegmentsFetched` event.

File: fluxc/src/main/java/org/wordpress/android/fluxc/model/vertical/VerticalModel.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.wordpress.android.fluxc.model.vertical
+
+import com.yarolegovich.wellsql.core.Identifiable
+import com.yarolegovich.wellsql.core.annotation.Column
+import com.yarolegovich.wellsql.core.annotation.PrimaryKey
+import com.yarolegovich.wellsql.core.annotation.Table
+
+@Table
```
Comment (malinajirka) [inline fluxc/src/main/java/org/wordpress/android/fluxc/model/vertical/VerticalModel.kt]: We should probably delete the @Table annotation.

File: fluxc/src/main/java/org/wordpress/android/fluxc/model/vertical/VerticalSegmentModel.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package org.wordpress.android.fluxc.model.vertical
+
+import com.yarolegovich.wellsql.core.Identifiable
+import com.yarolegovich.wellsql.core.annotation.Column
+import com.yarolegovich.wellsql.core.annotation.PrimaryKey
+import com.yarolegovich.wellsql.core.annotation.Table
+
+@Table
```
Comment (malinajirka) [inline fluxc/src/main/java/org/wordpress/android/fluxc/model/vertical/VerticalSegmentModel.kt]: We should probably delete the @Table annotation.

**Comment by malinajirka** [comment]
Thanks @oguzkocer ! I've left one comment, but otherwise LGTM.

**Comment by malinajirka** [comment]
Thanks @oguzkocer ! LGTM

>We should also look into renaming these models and removing the Model postfix since

👍

**Pull request merged**